pkg/kubelet/dockertools/docker.go

```diff
@@ -144,23 +145,32 @@ func filterHTTPError(err error, image string) error {
 }
 
 func (p dockerPuller) Pull(image string, secrets []api.Secret) error {
-	// If no tag was specified, use the default "latest".
-	imageID, tag := parsers.ParseImageName(image)
+	if parsers.GetImageTag(image) == "" {
+		// If no tag is specified, apply the default tag.
```

Comment:
IIUC, you need the "image:tag" string to pull the image. We can either
 1. Assume the input image string is always in the desired "image:tag" format. This is probably always true because we apply default tag (":latest") to images in the api.
 2. Call `ParseImageName` again, and assemble the two parts ourselves.

Am I missing something?